CommunityToolkit.Graph.Uwp/Extensions/ElementExtensions.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using CommunityToolkit.Authentication;
+using Windows.UI.Xaml;
+
+namespace CommunityToolkit.Graph.Uwp
+{
+    /// <summary>
+    /// Extensions on FrameworkElement for responding to authentication changes from XAML.
+    /// </summary>
+    public class ElementExtensions : DependencyObject
+    {
+        private static readonly IList<FrameworkElement> _listeningElements = new List<FrameworkElement>();
```

Comment:
Wondering if we want a `ConcurrentDictionary` would be a tiny bit safer here? ([reference for another use case in the Toolkit already](https://github.com/windows-toolkit/WindowsCommunityToolkit/blob/63cbb4a51bdef59083ccfc86bbcba41f966d0027/Microsoft.Toolkit.Uwp/Extensions/DispatcherQueueTimerExtensions.cs#L16))